barretenberg/cpp/src/barretenberg/ecc/scalar_multiplication/bitvector.hpp

```diff
@@ -21,7 +21,7 @@ class BitVector {
 
     BB_INLINE void set(size_t index, bool value) noexcept
     {
-        ASSERT(index < num_bits_);
+        BB_ASSERT_LT_DEBUG_ONLY(index, num_bits_);
```

Comment:
Since this method is `noexcept`, we can't throw. Should we keep it debug only or make it abort on release builds?

Reply:
hm i dont really see the value of us marking things noexcept when they can abort